webv2/uniqueid/uniqueid_assignment.go

```diff
@@ -0,0 +1,219 @@
+	"github.com/cloudspannerecosystem/harbourbridge/schema"
+	"github.com/cloudspannerecosystem/harbourbridge/spanner/ddl"
+	"github.com/cloudspannerecosystem/harbourbridge/webv2/session"
+)
+
+// AssignUniqueId to handles  cascading effect in UI.
+// Its iterate over source and spanner schema
+// and assign id to table and column.
+func AssignUniqueId(conv *internal.Conv) {
+
+	for sourcetablename, sourcetable := range conv.SrcSchema {
+
+		for spannertablename, spannertable := range conv.SpSchema {
+
+			if sourcetablename == spannertablename {
+
+				tableuniqueid := GenerateTableId()
+				sourcetable.Id = tableuniqueid
+				spannertable.Id = tableuniqueid
+
+				for spannercolumnname, spannercolumn := range spannertable.ColDefs {
+
+					if spannercolumn.Name == "synth_id" {
+
+						columnuniqueid := GenerateColumnId()
+						spannercolumn.Id = columnuniqueid
+						conv.SpSchema[spannertablename].ColDefs[spannercolumnname] = spannercolumn
+					}
+
+				}
+
+				for sourcecolumnname, sourcecolumn := range sourcetable.ColDefs {
+
+					for spannercolumnname, spannercolumn := range spannertable.ColDefs {
+
+						if sourcecolumn.Name == spannercolumn.Name {
+
+							columnuniqueid := GenerateColumnId()
+							sourcecolumn.Id = columnuniqueid
+							spannercolumn.Id = columnuniqueid
+
+							conv.SrcSchema[sourcetablename].ColDefs[sourcecolumnname] = sourcecolumn
+							conv.SpSchema[spannertablename].ColDefs[spannercolumnname] = spannercolumn
+
+							break
+						}
+					}
+
+				}
+
+				for sourceforeignkeyindex, sourceforeignkey := range sourcetable.ForeignKeys {
+
+					for spannerforeignkeyindex, spannerforeignkey := range spannertable.Fks {
+
+						if sourceforeignkey.Name == spannerforeignkey.Name {
+
+							foreignkeyid := GenerateForeignkeyId()
+
+							sourceforeignkey.Id = foreignkeyid
+							spannerforeignkey.Id = foreignkeyid
+
+							conv.SrcSchema[sourcetable.Name].ForeignKeys[sourceforeignkeyindex] = sourceforeignkey
+							conv.SpSchema[spannertable.Name].Fks[spannerforeignkeyindex] = spannerforeignkey
+						}
+
+					}
+				}
+
+				for sourcei, sourceindexes := range sourcetable.Indexes {
+
+					for spanneri, spannerindexes := range spannertable.Indexes {
+
+						if sourceindexes.Name == spannerindexes.Name {
+
+							indexesid := GenerateIndexesId()
+
+							sourceindexes.Id = indexesid
+							spannerindexes.Id = indexesid
+
+							conv.SrcSchema[sourcetable.Name].Indexes[sourcei] = sourceindexes
+							conv.SpSchema[spannertable.Name].Indexes[spanneri] = spannerindexes
+
+						}
+
+					}
+				}
+
+				updateSpannerTableIndexKeyOrder(spannertable)
+				updateSourceTableIndexKeyOrder(sourcetable)
+
+				conv.SrcSchema[sourcetablename] = sourcetable
+				conv.SpSchema[spannertablename] = spannertable
+				break
+			}
+		}
+	}
+
+}
+
+// updateSpannerTableIndexKeyOrder Update Primary Key Order as columnId.
+func updateSpannerTableIndexKeyOrder(spannertable ddl.CreateTable) {
+
```

Comment:
in case of two columns as primary key it is allowing 0 and 2 as orders to valid input